source/material_model/composition_reaction.cc

```diff
@@ -241,7 +213,7 @@ namespace aspect
 
       // Declare dependencies on solution variables
       this->model_dependence.viscosity = NonlinearDependence::none;
-      this->model_dependence.density = NonlinearDependence::none;
+      this->model_dependence.density = NonlinearDependence::temperature | NonlinearDependence::compositional_fields;
```

Comment:
Do you want to ask the EoS for the dependency? Maybe this whole kind of computation also needs to go into the EoS

Reply:
I don't think these dependencies are very useful. I can't see why anyone would run a model where the density does not depend on at least the temperature or the composition, and all other material models simply assume that this is the case, instead of checking the parameter values. I am not even sure where this is used...

Conceptually, this is something that belongs to the equation of state, but I simply don't think it's worth the effort, and it would add more code.

Reply:
Fair enough :-)